jumpscale/packages/vdc_dashboard/bottle/controller.py

```diff
@@ -0,0 +1,275 @@
+    """
+    vdc_dict = _get_vdc_dict()
+
+    return HTTPResponse(
+        j.data.serializers.json.dumps(vdc_dict), status=200, headers={"Content-Type": "application/json"}
+    )
+
+
+@app.route("/api/controller/node/list", method="POST")
+@controller_authorized()
+def list_nodes():
+    """
+    request body:
+        password
+
+    Returns:
+        kubernetes: string
+    """
+    vdc_dict = _get_vdc_dict()
+
+    return HTTPResponse(
+        j.data.serializers.json.dumps(vdc_dict["kubernetes"]), status=200, headers={"Content-Type": "application/json"}
+    )
+
+
+@app.route("/api/controller/node/add", method="POST")
+@controller_authorized()
+def add_node():
+    """
+    request body:
+        password
+        flavor
+
+    Returns:
+        wids: list of wids
+    """
+    data = j.data.serializers.json.loads(request.body.read())
+    vdc_password = data.get("password")
+    node_flavor = data.get("flavor")
+    if not all([node_flavor]):
+        abort(400, "Error: Not all required params were passed.")
+
+    # check stellar service
+    if not j.clients.stellar.check_stellar_service():
+        abort(400, "Stellar service currently down")
+
+    if node_flavor.upper() not in VDC_SIZE.K8SNodeFlavor.__members__:
+        abort(400, "Error: Flavor passed is not supported")
+    node_flavor = node_flavor.upper()
+
+    vdc = get_vdc()
+    vdc.load_info()
+    deployer = vdc.get_deployer(password=vdc_password)
+    capacity_check, farm_name = vdc.check_capacity_available(node_flavor)
+    if not capacity_check:
+        abort(400, f"There's no enough capacity in farm {farm_name} for kubernetes node of flavor {node_flavor}")
+
+    # Payment
+    success, _, _ = vdc.show_external_node_payment(bot=None, size=node_flavor, public_ip=False)
+    if not success:
+        abort(400, "Not enough funds in prepaid wallet to add node")
+
+    old_wallet = deployer._set_wallet(vdc.prepaid_wallet.instance_name)
+    try:
+        wids = deployer.add_k8s_nodes(node_flavor, public_ip=False)
+        deployer._set_wallet(old_wallet)
+        return HTTPResponse(
+            j.data.serializers.json.dumps(wids), status=201, headers={"Content-Type": "application/json"}
+        )
+    except Exception as e:
+        abort(400, f"failed to add nodes to your cluster. due to error {str(e)}")
+
+
+@app.route("/api/controller/node/delete", method="POST")
+@controller_authorized()
+def delete_node():
+    """
+    request body:
+        password
+        wid
+
+    Returns:
+        status
+    """
+    data = j.data.serializers.json.loads(request.body.read())
+    vdc_password = data.get("password")
+    wid = data.get("wid")
+    if not all([wid]):
+        abort(400, "Error: Not all required params were passed.")
+
+    vdc = get_vdc()
+    vdc.load_info()
+    deployer = vdc.get_deployer(password=vdc_password)
+
+    try:
+        deployer.delete_k8s_node(wid)
+    except j.exceptions.Input:
+        abort(400, "Error: Failed to delete workload")
```

Comment:
I think many exceptions could happen here

```suggestion
    except Exception as e:
        j.logger.error("Error: Failed to delete workload due to {str(e)}")
        abort(500, "Error: Failed to delete workload")
```